### PR TITLE
feat(core): gutenberg block support

### PR DESCRIPTION
**Pull request by bdbch**
This is a draft pull request for Gutenberg support as mentioned in #17. It's still WIP as I'm updating it. Help is appreciated!

### CODE REVIEW

File: lib/Init.php [hunk 1]
```diff
@@ -18,6 +18,9 @@ public static function initTheme()
         // Set to true to load all assets from a CDN if there is one specified
         Asset::loadFromCdn(false);
 
+        // Set to true to use the new Gutenberg editor in combination with Flynt components
+        Defaults::useGutenberg(true);
```
Comment (bdbch) [inline lib/Init.php]: I wasn't sure if this should be handled in the Default class or API class. any thoughts? I also thought about having a Gutenberg class and use it to configure and define variables on how Gutenberg is configured in Flynt.

File: lib/Utils/TwigExtensionFlynt.php [hunk 2]
```diff
@@ -25,11 +26,21 @@ public function getFunctions()
 
     public function renderFlexibleContent(Twig_Environment $env, $context, $fields, $withContext = true, $ignoreMissing = false, $sandboxed = false)
     {
-        $output = '';
-        foreach ((empty($fields) ? [] : $fields) as $field) {
-            $output .= $this->renderComponent($env, $context, ucfirst($field['acf_fc_layout']), $field, $withContext, $ignoreMissing, $sandboxed);
+        if (!Defaults::$useGutenberg) {
```
Comment (bdbch) [inline lib/Utils/TwigExtensionFlynt.php]: Any thoughts on this one? As you can see I rely on `the_content` for Gutenberg.

Has some benefits f.e. RSS support, post excerpts, etc. but won't allow you to have multiple flexibleContent renderers on one page.

Any thoughts @domtra?

File: lib/Api.php [hunk 3]
```diff
@@ -82,4 +84,63 @@ public static function registerHooks()
             );
         }, 10, 2);
     }
+
+    public static function registerBlocks($layouts)
+    {
+        // create block and field group for block
+        if (!function_exists('acf_register_block_type')) return null;
```
Comment (bdbch) [inline lib/Api.php]: I think this check could be done right before useGutenberg is defined. If the function does not exist it should just disable gutenberg again, will fix later.

**Comment by bdbch** [comment]
Does anybody want to take a look?

**Comment by cbirdsong** [comment]
@bdbch Has this stalled out?

**Comment by bdbch** [comment]
@cbirdsong I could still pick it up but it's stale for now. You may give it a try if you want to!

**Comment by cbirdsong** [comment]
@bdbch I assume even with this Flynt block data is stored in JSON blobs and not actual HTML, much like standard ACF blocks?

**Comment by bdbch** [comment]
I used ACF Gutenberg blocks for this:
https://www.advancedcustomfields.com/blog/acf-5-8-introducing-acf-blocks-for-gutenberg/

They don't save the data as html but as a dataset as you'd expect.

**Comment by steffenbew** [comment]
Decided not to support 2 approaches (standard ACF fields + ACF blocks) side by side. See #17 for further thoughts about why we don't support Gutenberg yet.

**Pull request closed without merging**